### PR TITLE
Rln-relay integration into chat2

### PR DESCRIPTION
**Pull request by staheri14**
Addresses the following items of #830:
- [x] Adding RLN-Relay CLI configs to chat2
- [x] Mounting RLN-Relay inside chat2
- [x] Appending rate limit proofs to chat2 messages
- [x] Testing in nim

In this PR, I also added another feature: Rln-relay can be activated for a specific combination of pusub topic and content topic supplied by the user via command-line arguments. This feature is enabled both in chat2 and wakunode2. 
For dogfooding using chat2, we will use the default pubsub topic i.e., `/waku/2/default-waku/proto` however we may set a different content topic e.g.,`waku/2/rln-relay/proto`  or stick to the default `waku/2/huilong/proto` (will using the default content topic `waku/2/huilong/proto`  cause any issue? @jm-clius)

# Next
As the next step (and after performing some local tests by running rln-relay enabled chat2 clients and making them communicate), I will create a tutorial for dogfooding.

### CODE REVIEW

File: waku/v2/node/wakunode2.nim [hunk 1]
```diff
@@ -468,26 +468,29 @@ proc mountStore*(node: WakuNode, store: MessageStore = nil, persistMessages: boo
...
+        # check the contentTopic
+        if (wakumessage.contentTopic != "") and (wakumessage.contentTopic != contentTopic):
+          info "content topic did not match:", contentTopic=wakumessage.contentTopic, payload=string.fromBytes(wakumessage.payload)
+          return pubsub.ValidationResult.Accept
+        # validate the message
+        let validationRes = node.wakuRlnRelay.validateMessage(wakumessage)
         case validationRes:
           of Valid:
-            info "message validity is verified, relaying:", wakumessage=wakumessage
+            info "message validity is verified, relaying:", wakumessage=wakumessage, payload=string.fromBytes(wakumessage.payload)
```
Comment (jm-clius) [inline waku/v2/node/wakunode2.nim:487]: Does this mean that with RLN enabled we'll log at `info` level for each message? Per-message logs should preferably be at `TRC` level to help keep logs useful.

**Review by jm-clius — COMMENTED** [review]
I've added a couple of comments, but it's exciting that we could soon test rln in chat2! As for your question about using the `huilong` content topic - I don't see any issues.

File: examples/v2/chat2.nim [hunk 2]
```diff
@@ -217,8 +233,18 @@ proc publish(c: Chat, line: string) =
...
   else:
     # No payload encoding/encryption from Waku
-    let message = WakuMessage(payload: chat2pb.buffer,
+    var message = WakuMessage(payload: chat2pb.buffer,
       contentTopic: c.contentTopic, version: 0)
+    when defined(rln):
+      if  not isNil(c.node.wakuRlnRelay):
+        # for future version when we support more than one rln protected content topic, 
+        # we should check the message content topic as well
+        let success = c.node.wakuRlnRelay.appendRLNProof(message, epochTime())
```
[the remark this answers is not in the document]
Comment (staheri14) [reply examples/v2/chat2.nim:242]: Sure, I'm working on it.

File: waku/v2/node/wakunode2.nim (hunk 1 above)
Comment (staheri14) [reply waku/v2/node/wakunode2.nim:487]: Sure, I'll change the log level to trace or debug

File: examples/v2/chat2.nim (hunk 2 above)
[the remark this answers is not in the document]
Comment (staheri14) [reply examples/v2/chat2.nim]: done in https://github.com/status-im/nim-waku/pull/835/commits/e6cc13364a4b297871992d8e139fac4d36929e76

File: waku/v2/node/wakunode2.nim (hunk 1 above)
Comment (staheri14) [reply waku/v2/node/wakunode2.nim]: done in https://github.com/status-im/nim-waku/pull/835/commits/ea4c9a6a9df9a27aae99e1dfec0279106c754b0b

File: waku/v2/node/config.nim [hunk 3]
```diff
@@ -83,8 +84,13 @@ type
 
     rlnRelayPubsubTopic* {.
       desc: "the pubsub topic for which rln-relay gets enabled",
-      defaultValue: "waku/2/rlnrelay/proto"
+      defaultValue: "/waku/2/default-waku/proto"
       name: "rln-relay-pubsub-topic" }: string
+
+    rlnRelayContentTopic* {.
+      desc: "the pubsub topic for which rln-relay gets enabled",
+      defaultValue: "waku/2/huilong/proto"
```
Comment (jm-clius) [inline waku/v2/node/config.nim:92]: I understand that we may want to default to the `huilong` content topic for `chat2`, but wouldn't it make sense for the default behaviour of a `wakunode2` to enable rln for "all" content topics (I think that corresponds to an empty string `""`)?
The `huilong` content topic is `/toy-chat/2/huilong/proto`

File: waku/v2/node/wakunode2.nim [hunk 4]
```diff
@@ -468,26 +468,29 @@ proc mountStore*(node: WakuNode, store: MessageStore = nil, persistMessages: boo
...
+        # check the contentTopic
+        if (wakumessage.contentTopic != "") and (wakumessage.contentTopic != contentTopic):
+          trace "content topic did not match:", contentTopic=wakumessage.contentTopic, payload=string.fromBytes(wakumessage.payload)
+          return pubsub.ValidationResult.Accept
+        # validate the message
+        let validationRes = node.wakuRlnRelay.validateMessage(wakumessage)
         case validationRes:
           of Valid:
-            info "message validity is verified, relaying:", wakumessage=wakumessage
+            trace "message validity is verified, relaying:", wakumessage=wakumessage, payload=string.fromBytes(wakumessage.payload)
```
Comment (jm-clius) [inline waku/v2/node/wakunode2.nim:487]: Would it make sense to add a metric here that counts all the different types of validation responses?

**Review by jm-clius — APPROVED** [review]
LGTM. My only comments are about adding metrics (would be nice to monitor validation results on a dashboard :D) and whether it would make sense to have some kind of "open" content topic configuration for RLN which will enable RLN on all content topics? That way the default `wakunode2` config does not need to know about `huilong` (which was a construct for the toy-chat). Later on it may even be necessary to somehow provide an input lists of RLN-managed content topics.

File: waku/v2/node/wakunode2.nim (hunk 4 above)
Comment (staheri14) [reply waku/v2/node/wakunode2.nim:487]: Thanks, for the metrics, I am gonna open another PR in which I will also include the validation responses.

File: waku/v2/node/config.nim (hunk 3 above)
Comment (staheri14) [reply waku/v2/node/config.nim]: > The huilong content topic is /toy-chat/2/huilong/proto.

Had it fixed locally but forgot to push, thanks. Please see https://github.com/status-im/nim-waku/pull/835/commits/44bef3c4d27e8a2befabeff6ebc6063f7812e014

Also added support for any contentTopic for a specific pubsub topic, please see https://github.com/status-im/nim-waku/pull/835/commits/10ecc77a86b46be6bf9ae9ded5dea93379fa489a

**Comment by staheri14** [comment]
> That way the default wakunode2 config does not need to know about huilong (which was a construct for the toy-chat).

@jm-clius  Despite the support added for any content topic, I assume we do not want to spam-protect all the content topics of the default waku network in the dogfooding session, other topics may be used by other applications and it is not desirable to enforce spam protection over them, WDYT? once the topic validator is mounted on the test fleets for `/waku/2/default-waku/proto`, then the spam protection will affect any message published on `/waku/2/default-waku/proto`

**Pull request merged**